perun/collect/trace/parse.py

```diff
@@ -0,0 +1,265 @@
+    :param dict func: the function probe specifications
+    :param dict static: the static probe specifications as dictionaries
+    :param kwargs: additional parameters
+    :return object: the generator object that produces dictionaries representing the resources
+    """
+    WD.info('Transforming the raw performance data into a perun profile format')
+    trace_stack, sequence_map = {}, {}
+
+    with open(data_file, 'r') as trace:
+        cnt = 0
+        line = ''
+        try:
+            # Initialize just in case the trace doesn't have 'begin' statement
+            trace_stack, sequence_map = _init_stack_and_map(func, static)
+            for cnt, line in enumerate(trace):
+                # File starts or ends
+                if line.startswith('begin '):
+                    # Initialize stack and map
+                    trace_stack, sequence_map = _init_stack_and_map(func, static)
+                    continue
+                elif line.startswith('end '):
+                    return
+                # Parse the line into the _TraceRecord tuple
+                record = _parse_record(line)
+                # Process the record
+                resource = _process_record(record, trace_stack, sequence_map, static,
+                                           kwargs['global_sampling'])
+                if resource:
+                    resource['workload'] = kwargs.get('workload', ' '.join(kwargs['workload']))
+                    yield resource
+            WD.info('Data to profile transformation finished')
+        except Exception:
+            WD.info('Error while parsing the raw trace record')
+            # Log the status in case of unhandled exception
+            WD.log_trace_stack(line, cnt, trace_stack)
+            raise
+
+
+def _init_stack_and_map(func, static):
+    """Initializes the data structures of function and static stacks for the trace parsing
+
+    :param dict func: the function probes
+    :param dict static: the static probes
+    :return tuple: initialized trace stack and sequence map
+    """
+
+    def default_sequence_records(records):
+        """ Initializes the default dictionaries for the sequence map
+
+        :param iterable records: the iterable records
+        :return dict: the resulting dictionary
+        """
+        return {record['name']: {'seq': 0, 'sample': record['sample']} for record in records}
+
+    # func: thread -> stack (stack list, faults list)
+    # static: thread -> name -> stack
+    trace_stack = {
+        'func': collections.defaultdict(lambda: ([], [])),
+        'static': collections.defaultdict(lambda: collections.defaultdict(list))
+    }
+    # name -> sequence values
+    sequence_map = {
+        'func': default_sequence_records(func.values()),
+        'static': default_sequence_records(static.values())
+    }
+    return trace_stack, sequence_map
+
+
+# TODO: this should be used only after symbol cross-compare is functional
```

Comment:
Suggestoin: Is this needed?

Reply:
It could be useful in the future, mainly for the user. Thus it serves as a remainder of potential future work.